### DISCUSSION TITLE
Default flushMode for em.transactional

### DISCUSSION THREAD

**Discussion by dirslashls** [mikro-orm/mikro-orm#4785] [General]
I ran into an issue of `findOneOrFail` trying to flush within the transactional callback and it turns out if the flushMode is not specified (at various places as per the docs), it defaults to AUTO which triggers a commit. I managed to workaround this by using COMMIT as the flush mode. However, I am wondering if `em.transactional` should have default flushMode as COMMIT (even if the Orm Options is set to AUTO or ALWAYS) as that is what is naturally assumed, that the entire callback is a single commit unit.

**Comment by B4nan** [comment]
>  that the entire callback is a single commit unit.

It is still a single unit, the auto flush will create a child transaction when it happens inside `em.transactional()`, it won't commit the one outside.

**Comment by dirslashls** [comment]
Thanks @B4nan. In my case, I wasn't expecting child transactions so flushMode of COMMIT was appropriate for my use case. Just curious how this child transactions concept implementation works for databases that don't support nested transactions.
